integration/spark/src/main/scala/org/apache/carbondata/spark/util/GlobalDictionaryUtil.scala

```diff
@@ -742,7 +742,8 @@ object GlobalDictionaryUtil extends Logging {
    */
   def generateGlobalDictionary(sqlContext: SQLContext,
                                carbonLoadModel: CarbonLoadModel,
-                               hdfsLocation: String): Unit = {
+                               hdfsLocation: String,
+                               dataFrame: Option[DataFrame] = None): Unit = {
```

Comment:
Can you also add a option in CarbonOption to cache the dataframe? So that user can decide whether to cache it. The name of the option can be `cache`, default is `false`